api/controllers/v2/cache.ts

```diff
@@ -0,0 +1,185 @@
+ * In the latter case, we should receive an event to build all doc site caches.
+ * @param {APIGatewayEvent} event An event object that comes from either a webhook payload or from the custom GitHub Action for the docs-worker pool.
+ *
+ * In either scenario, the body should contain an array of RepoInfo objects.
+ * @returns {Promise<APIGatewayProxyResult>}
+ */
+export async function rebuildCacheHandler(event: APIGatewayEvent): Promise<APIGatewayProxyResult> {
+  if (!event.body) {
+    const errorMessage = 'Error! No body found in event payload.';
+    console.error(errorMessage);
+    return {
+      statusCode: 400,
+      body: errorMessage,
+    };
+  }
+
+  const rebuildRequest = JSON.parse(event.body);
+
+  if (!isRebuildRequest(rebuildRequest)) {
+    const errorMessage = 'Error! Invalid rebuild request.';
+    console.error(errorMessage);
+    return {
+      statusCode: 400,
+      body: errorMessage,
+    };
+  }
+
+  try {
+    await runCacheRebuildJob(rebuildRequest);
+    return {
+      statusCode: 200,
+      body: 'Cache rebuild job successfully created',
+    };
+  } catch (e) {
+    console.error(e);
+    return {
+      statusCode: 500,
+      body: 'Error occurred when starting cache rebuild job',
+    };
+  }
+}
+
+/**
+ * This is for the GitHub webhooks. The GitHub webhooks will be used by individual doc sites to rebuild the cache if
+ * the snooty.toml file is modified.
+ * @param {APIGatewayEvent} event GitHub webhook push event. Body should be a PushEvent type.
+ * @returns {Promise<APIGatewayProxyResult>}
+ */
+export async function rebuildCacheGithubWebhookHandler(event: APIGatewayEvent): Promise<APIGatewayProxyResult> {
+  if (!event.body) {
+    const errorMessage = 'Error! No body found in event payload.';
+    console.error(errorMessage);
+    return {
+      statusCode: 400,
+      body: errorMessage,
+    };
+  }
+
+  let body: PushEvent;
+  try {
+    body = JSON.parse(event.body) as PushEvent;
+  } catch (e) {
+    console.log('[TriggerBuild]: ERROR! Could not parse event.body', e);
+    return {
+      statusCode: 502,
+      headers: { 'Content-Type': 'text/plain' },
+      body: ' ERROR! Could not parse event.body',
+    };
+  }
+
+  const repoOwner = body.repository.owner.login;
+  const repoName = body.repository.name;
+
+  // Checks the commits to see if there have been changes made to the snooty.toml file.
+  const snootyTomlChanged = body.commits.some(
+    (commit) =>
+      commit.added.some((fileName) => fileName === 'snooty.toml') ||
+      commit.removed.some((fileName) => fileName === 'snooty.toml') ||
+      commit.modified.some((fileName) => fileName === 'snooty.toml')
+  );
+
+  if (!snootyTomlChanged) {
+    return { statusCode: 202, body: 'snooty.toml has not changed, no need to rebuild cache' };
+  }
+
+  const ref = body.ref;
+  // For webhook requests, this should only run on the primary branch, and if the repository belongs to the 10gen or mongodb orgs.
```

Comment:
Does this mean we'll have to add these new webhooks to every individual docs repo?

Reply:
Yes, this is correct